src/config/logger.ts

```diff
@@ -1,24 +1,62 @@
-// Configuration for `winston` package.
 import hasAnsi from 'has-ansi'
 import omit from 'lodash/omit'
+import logform from 'logform'
+import path from 'path'
+import tripleBeam from 'triple-beam'
 import { inspect } from 'util'
 import { v4 as uuidv4 } from 'uuid'
-import { format, LoggerOptions, loggers, transports } from 'winston'
+import { format, Logger, LoggerOptions, loggers, transports } from 'winston'
 import WinstonCloudWatch from 'winston-cloudwatch'
 
 import defaults from './defaults'
 
+// Params to enforce the logging format.
+type CustomLoggerParams = {
+  message: string
+  meta: {
+    action: string
+    [other: string]: any
+  }
+  error?: Error
+}
+
 // Cannot use config due to logger being instantiated first, and
 // having circular dependencies.
 const isDev = ['development', 'test'].includes(process.env.NODE_ENV)
 const customCloudWatchGroup = process.env.CUSTOM_CLOUDWATCH_LOG_GROUP
 const awsRegion = process.env.AWS_REGION || defaults.aws.region
 
-// Config to make winston logging like console logging, allowing multiple
-// arguments.
+// A variety of helper functions to make winston logging like console logging,
+// allowing multiple arguments.
 // Retrieved from
 // https://github.com/winstonjs/winston/issues/1427#issuecomment-583199496.
 
+/**
+ * Hunts for errors in the given object passed to the logger.
+ * Assigns the `error` key the found error.
+ */
+const errorHunter = logform.format((info) => {
```

Comment:
This seems to be from [here](https://gist.github.com/stieg/3098a4c79471e659b7ec55f49ded7c17) - could you add some docs explaining what this does? it's very opaque to me

Reply:
It does what it seems to do, checks out each key of the given `info` object, filters and finds if there are any `Error` classes in the values, and assigns `info.error` that filtered `Error`, so the pipeline can tell that there is an error and format it accordingly.

Reply:
>It does what it seems to do
lol I'm more of getting at why this is necessary. even a one-liner like "just in case there are hidden error objects in the log info" would help. and after a few minutes of googling I can't find any clear documentation on what `tripleBeam.SPLAT` is or does, so some explanation would be useful there as well.

Reply:
Gotcha. Winston relies on their [`logform`](https://github.com/winstonjs/logform) package for formatting, which in turn uses `triplebeam`. This is abstraction hell at its finest :P